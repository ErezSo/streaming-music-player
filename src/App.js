import React, { Component } from "react";
import { Navbar, ArtistsFeed } from "./components";

class App extends Component {
  render() {
    return (
      <div className="App">
        <Navbar />
        <ArtistsFeed />
      </div>
    );
  }
}

export default App;
