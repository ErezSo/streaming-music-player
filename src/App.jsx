import React, { Component } from "react";
import { BrowserRouter, Route } from "react-router-dom";
import { ArtistsFeed, Navbar, SongPage } from "./components";

class App extends Component {
  state = {
    artists: [],
    song: ""
  };

  componentDidMount() {
    return fetch(
      "https://api-v2.hearthis.at/feed/?type=popular&page=1&count=20"
    )
      .then(res => res.json())
      .then(data => this.setState({ artists: data }))
      .catch(error => {
        throw error;
      });
  }

  playSong = song => this.setState({ song });

  render() {
    const { artists } = this.state;
    return (
      <BrowserRouter>
        <div>
          <Navbar song={this.state.song} />
          <div className="container-fluid">
            <Route
              exact
              path="/"
              render={() => <ArtistsFeed artists={artists} />}
            />
            <Route
              exact
              path="/songs/:id"
              render={match => (
                <SongPage
                  artists={artists}
                  playSong={this.playSong}
                  match={match}
                />
              )}
            />
          </div>
        </div>
      </BrowserRouter>
    );
  }
}

export default App;
