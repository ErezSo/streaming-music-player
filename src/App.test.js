import React from "react";
import { shallow } from "enzyme";
import App from "./App";

describe("App", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it("renders", () => {
    expect(wrapper).toHaveLength(1);
  });

  it("renders correctly", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
