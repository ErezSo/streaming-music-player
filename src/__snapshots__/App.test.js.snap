// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
"<BrowserRouter>
  <div>
    <Navbar song=\\"\\" />
    <div className=\\"container-fluid\\">
      <Route exact={true} path=\\"/\\" render={[Function: render]} />
      <Route exact={true} path=\\"/songs/:id\\" render={[Function: render]} />
    </div>
  </div>
</BrowserRouter>"
`;
